src/Features/CSharp/Portable/CodeRefactorings/NodeSelectionHelpers.cs

```diff
@@ -53,7 +55,10 @@ internal static async Task<ImmutableArray<SyntaxNode>> GetSelectedDeclarationsOr
                 // pick up on keywords before the declaration, such as "public static int".
                 // We could potentially use it for every case if that behavior changes
                 var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                return await CSharpSelectedMembers.Instance.GetSelectedMembersAsync(tree, span, allowPartialSelection: true, cancellationToken).ConfigureAwait(false);
+                var members = await CSharpSelectedMembers.Instance.GetSelectedMembersAsync(tree, span, allowPartialSelection: true, cancellationToken).ConfigureAwait(false);
+                return members.Any(m => m is GlobalStatementSyntax or IncompleteMemberSyntax)
+                    ? ImmutableArray<SyntaxNode>.Empty
+                    : members;
```

Comment:
Why return empty array instead of just filtering the unwanted members (`members.Where(m => m is not (GlobalStatementSyntax or IncompleteMemberSyntax))`)?

Is there any test that can demonstrate the difference between the two approaches? My feeling is that we should offer the refactoring and just skip the bad declarations if that's possible.

Reply:
My general thinking is that 1) in the case of a global statement, everything in the file should also be a global statement, so nothing should be available, and 2) in case of an incomplete member, the incompleteness may result in a parse that gives a member that we are not expecting. I don't have a test for this (I can try to write one), but I'm thinking in the case of `[|publicc int foo;|]` Could give a valid member of `foo`, and an incomplete member of `publicc`. The problem would be that the `publicc` wouldn't be attached and so the user might expect that a refactored member `foo` would be public, when it actually isn't. I definitely see the positives for simply filtering as well.

Reply:
yeah, it's an interesting choice.  i can see arguments in both directions.  at teh very least, this would be a good piece of information to add as a comment here as to why we think that we should fail out the moment we have one of these.

Reply:
In `CSharpMoveStaticMembersTests`, `TestSelectMalformedField_NoAction3` is an example of this. We get one malformed field called `st`, then another one called `TestField`. In this case neither are static so they don't trigger anyways, but I'll write a better version of the test, and incorporate for other refactoring tools.

Reply:
I like this current behavior : )